### ISSUE TITLE
[no-implied-eval] False positive when importing symbol with name Function

### ISSUE THREAD

**Issue by marcogrcr** [typescript-eslint/typescript-eslint#2338]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-implied-eval": "error"
  }
}
```

```TS
import { Function } from "@aws-cdk/aws-lambda";

new Function(scope, "MyLambdaFunction", { /* ... */ });
```

**Expected Result** The rule detects that the imported `Function` is not global `Function` (either from the `import` or the parameters).

**Actual Result** The following error is logged:

```
error    Implied eval. Do not use the Function constructor to create functions  @typescript-eslint/no-implied-eval
```

**Additional Info**

This `if` statement is evaluating to `true`:

https://github.com/typescript-eslint/typescript-eslint/blob/2d80c5155ffd122e9abc3906fd1099d8367689b3/packages/eslint-plugin/src/rules/no-implied-eval.ts#L121-L124

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.0` |
| `@typescript-eslint/parser`        | `3.6.0` |
| `TypeScript`                       | `3.9.6` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
wow - did they really call a part of their API `Function`? That seems so silly to have a name clash like that.

This isn't that hard to fix - we already do a similar check in `no-throw-literal` to ensure the type is the library type and not an imported type:

https://github.com/typescript-eslint/typescript-eslint/blob/817067b72b53c8f417f3c1c85e970ea82f0f6953/packages/eslint-plugin/src/rules/no-throw-literal.ts#L43-L51

Happy to accept a PR!

**Comment by soobing** [comment]
@bradzacher  I made PR 💗
